File: js/BeginnersSelection/05_SomeSums.js
// https://atcoder.jp/contests/abs/tasks/abc083_b

Object.prototype.let = function(f) {
  return f(this.valueOf());
}

require('fs')
  .readFileSync('/dev/stdin', 'utf8')
  .split(' ')
  .map(x => +x)
  .let(([n, a, b]) =>
    [...Array(n).keys()].map(x => x + 1)
      .filter(i =>
        i.toString()
          .split('')
          .map(x => +x)
          .reduce((a, x) => a + x)
          .let(x => a <= x && x <= b)
      )
  )
  .reduce((a, x) => a + x)
  .let(console.log);

// https://atcoder.jp/contests/abs/submissions/21478051

File: rb/BeginnersSelection/05_SomeSums.rb
# https://atcoder.jp/contests/abs/tasks/abc083_b

class Object
  def let(f)
    f.call self
  end
end

gets
  .split(" ")
  .map(&:to_i)
  .let(->((n, a, b)) {
    (1..n)
      .filter { |i|
        i.to_s
          .split("")
          .map(&:to_i)
          .sum
          .let(->(x) { a <= x && x <= b })
      }
      .sum
  })
  .let(method(:puts))

# https://atcoder.jp/contests/abs/submissions/21519650
